benchmark_runner/common/template_operations/template_operations.py

```diff
@@ -120,7 +120,7 @@ def get_sub_dict(dictionary: dict, key: str, value: str):
                 template_file = f'{file_components[0]}_template{file_components[1]}'
             with open(os.path.join(workload_dir_path, 'internal_data', template_file)) as f:
                 template = Template(f.read())
-            answer[filename] = template.render(render_data)
+            answer[filename] = template.render(render_data) + "\n"
```

Comment:
Please use formatting instead of + (generate new objects in memory)
answer[filename] = f'{template.render(render_data)}\n'

Reply:
I don't think that the f-string and the `+` are much different in memory (considering that we're adding a single-char),
and I find the intend clearer with the concatenation (add EOL at EOF),
but I can change it to the f-string if you prefer

Reply:
I found what really bothered be with the f-string and that I wouldn't written in my code:
```
template.render(render_data) + "\n"
```
vs 
```
f"{template.render(render_data)}\n"
```
`template.render(render_data)` is a complex call (no side effect, but still not a trivial function), 
and it gets mixed with a formatting operation.

This would make more sense to me:
```
rendered_template = template.render(render_data)
answer = f"{rendered_template}\n"
```